### DISCUSSION TITLE
Is it poissible to pass some define to a file (or to all files of project) if I use compileCommands?

### DISCUSSION THREAD

**Discussion by tolezk** [microsoft/vscode-cpptools#9085] [Q&A] [answered]
I'm using this: https://github.com/amezin/vscode-linux-kernel to config the project. It has the next `c_cpp_properties.json`:
```
{
    "configurations": [
        {
            "name": "Linux",
            "cStandard": "c11",
            "intelliSenseMode": "linux-gcc-x64",
            "compileCommands": "${workspaceFolder}/compile_commands.json",
            "compilerPath": ""
        }
    ],
    "version": 4
}
```

I've tried to add such thing to this config:
```
"defines": [
    "__inline"
]
```

I expected, that IntelliSense would consider define '__inline' as applied to every file it parses. In practice, I got it doesn't.

`compilerPath` in my case may be set may be not.

So, is it possible somehow to pass a define I want to Intellisense?

**Comment by michelleangela** [comment, marked as answer]
This is not supported and the feature request for this is at https://github.com/microsoft/vscode-cpptools/issues/6909.
